### PR TITLE
Improve clang format

### PR DESCRIPTION
**Pull request by rhaschke**
This is an attempt to improve the results of clang-format. Particularly:
* increase penalty to break literal strings
* ~~increase penalty to break before first argument in function calls~~
* reduce penalty to put return type on its own line

I think the resulting code is much better readable.

### CODE REVIEW

**Comment by jonbinney** [comment]
+1 for this change.

**Comment by rhaschke** [comment]
Rebased / reapplied settings to resolve conflicts.

File: moveit_core/robot_model/src/joint_model_group.cpp [hunk 1]
```diff
@@ -326,9 +326,10 @@ void JointModelGroup::getVariableRandomPositionsNearBy(random_numbers::RandomNum
 {
   assert(active_joint_bounds.size() == active_joint_model_vector_.size());
   for (std::size_t i = 0; i < active_joint_model_vector_.size(); ++i)
-    active_joint_model_vector_[i]->getVariableRandomPositionsNearBy(
-        rng, values + active_joint_model_start_index_[i], *active_joint_bounds[i],
-        near + active_joint_model_start_index_[i], distance);
+    active_joint_model_vector_[i]->getVariableRandomPositionsNearBy(rng, values + active_joint_model_start_index_[i],
+                                                                    *active_joint_bounds[i],
+                                                                    near + active_joint_model_start_index_[i],
+                                                                    distance);
```
Comment (davetcoleman) [inline moveit_core/robot_model/src/joint_model_group.cpp]: i prefer the old version - this change is less readable IMO and adds another line.

File: moveit_core/robot_model/src/joint_model_group.cpp [hunk 2]
```diff
@@ -365,9 +367,10 @@ void JointModelGroup::getVariableRandomPositionsNearBy(random_numbers::RandomNum
                     boost::lexical_cast<std::string>(active_joint_model_vector_.size()) + ", but it is of size " +
                     boost::lexical_cast<std::string>(distances.size()));
   for (std::size_t i = 0; i < active_joint_model_vector_.size(); ++i)
-    active_joint_model_vector_[i]->getVariableRandomPositionsNearBy(
-        rng, values + active_joint_model_start_index_[i], *active_joint_bounds[i],
-        near + active_joint_model_start_index_[i], distances[i]);
+    active_joint_model_vector_[i]->getVariableRandomPositionsNearBy(rng, values + active_joint_model_start_index_[i],
+                                                                    *active_joint_bounds[i],
+                                                                    near + active_joint_model_start_index_[i],
+                                                                    distances[i]);
```
Comment (davetcoleman) [inline moveit_core/robot_model/src/joint_model_group.cpp]: same

**Comment by rhaschke** [comment]
I agree that this PR should only be merged _after_ other large pending PRs.
@davetcoleman, Obviously you don't mind to have a newline directly after the opening parenthesis of a function call. By reducing the penalty for this, we can easily get this behaviour back.
I think, the key contribution of this PR is to avoid breaking literal strings.

**Comment by rhaschke** [comment]
@davetcoleman Could you please have another look. I further tweaked the clang-format settings, such that the source lines you complained about are not touched anymore.
The reformatting only touches a few lines now. So there is no reason to hold this back anymore.

**Pull request merged**